File: HeartForCharity/HeartForCharity.Model/SearchObjects/ReviewSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HeartForCharity.Model.SearchObjects
{
    public class ReviewSearchObject : BaseSearchObject
    {
        public int? OrganisationProfileId { get; set; }
        public int? UserProfileId { get; set; }
        public int? MinRating { get; set; }
        public int? MaxRating { get; set; }
    }
}

File: HeartForCharity/UI/heartforcharity_desktop/lib/model/search_objects/review_search_object.dart
import 'package:heartforcharity_desktop/model/search_objects/base_search_object.dart';

class ReviewSearchObject extends BaseSearchObject {
  final int? organisationProfileId;
  final int? userProfileId;
  final int? minRating;
  final int? maxRating;

  ReviewSearchObject({
    super.fts,
    super.page,
    super.pageSize,
    super.includeTotalCount,
    super.retrieveAll,
    this.organisationProfileId,
    this.userProfileId,
    this.minRating,
    this.maxRating,
  });

  @override
  Map<String, dynamic> toMap() => {
        ...super.toMap(),
        if (organisationProfileId != null)
          'organisationProfileId': organisationProfileId,
        if (userProfileId != null) 'userProfileId': userProfileId,
        if (minRating != null) 'minRating': minRating,
        if (maxRating != null) 'maxRating': maxRating,
      };
}
